app/helpers/application_helper.rb

```diff
@@ -314,4 +314,12 @@ def render_tag_link(tag, filled: false, monochrome: false, classes: "")
 
     link_to(label, tag_path(tag), options)
   end
+
+  def creator_settings_form?
+    return unless User.with_role(:creator).any?
+
+    creator = User.with_role(:creator).first
+    FeatureFlag.enabled?(:creator_onboarding) && !creator.checked_code_of_conduct &&
```

Comment:
I think I would restructure this so we don't need to hit the DB unless the feature flag is enabled. This also makes it easier to later get rid of the flag as we just have to remove a single guard clause.

```ruby
return unless FeatureFlag.enabled?(:creator_onboarding)
return unless User.with_role(:creator).any?

creator = User.with_role(:creator).first
!creator.checked_code_of_conduct && !creator.checked_terms_and_conditions
```

Reply:
Good call! I've made this change.